internal/confmapprovider/discovery/discoverer.go

```diff
@@ -236,10 +238,19 @@ func (d *discoverer) createObserver(observerID config.ComponentID, cfg *Config)
 	if err = d.expandConverter.Convert(context.Background(), observerCfgMap); err != nil {
 		return nil, fmt.Errorf("error converting environment variables in %q config: %w", observerID.String(), err)
 	}
-	if err = observerCfgMap.Unmarshal(observerConfig); err != nil {
+
+	if err = config.UnmarshalExtension(observerCfgMap, observerConfig); err != nil {
 		return nil, fmt.Errorf("failed unmarshaling %s config: %w", observerID.String(), err)
 	}
 
+	if ce := d.logger.Check(zap.DebugLevel, "unmarshalled observer config"); ce != nil {
+		if c, e := yaml.Marshal(observerConfig); e != nil {
+			d.logger.Debug("failed unmarshalling observer config for logging", zap.Error(e))
```

Comment:
why not use `ce` here? also it failed marshaling to yaml, not unmarshaling, correct?

Reply:
The `ce` message is already `"unmarshalled observer config"` and in this failure path it's disregarded since the content isn't renderable. Corrected the wording, thanks